examples/roundtrip/roundtrip_common.hpp

```diff
@@ -69,6 +69,12 @@ class RoundTripListener: public dds::sub::NoOpDataReaderListener<RoundTripModule
     (void)_f(rd, _wr);
   }
 
+  void on_liveliness_changed(dds::sub::DataReader<RoundTripModule::DataType>&,
+        const dds::core::status::LivelinessChangedStatus& status) {
+    if (status.not_alive_count_change() > 0)
+      done = true;
+  }
+
```

Comment:
Why does the C++ API use a different condition to terminate then the C API? It seems that the C API terminates when the data gets disposed, where the C++ API terminates when the Writer disappears. Also in C the WriterDataLifcycle has its autodispose field set to FALSE, whereas in C++ it is set to TRUE. This means that in C you can instruct a Writer to write 10 samples of 10 bytes, gracefully exiting when done , and then restart it to write 10 additional samples of let's say 100 bytes, and continue doing this for ever increasing sizes. When done, you instruct ping to send a TERM message, in which case it disposes the data, after which pong will terminate itself gracefully. Up to that time, the pong application will keep on working to bounce back the samples produced by ping. In contrast, the C++, pong will terminate as soon as ping destructs its Writer, which means for the same scenario I not only need to restart ping, but restart pong as well. 
I would therefore like to propose the following:
- Set the autodispose field of WriterDataLifecycle to FALSE in C++.
- Remove the on_liveliness_changed handler in C++.
- When consuming samples, check its lifecycle and exit when it becomes DISPOSED.

Reply:
I will fix this